event_model/tests/test_em.py

```diff
@@ -481,6 +481,55 @@ def __call__(self, c, d):
         filler = event_model.Filler(reg)
 
 
+def test_rechunk_event_pages():
+
+    def event_page_gen(page_size, num_pages):
+        """
+        Generator event_pages for testing.
+        """
+        data_keys = ['x', 'y', 'z']
+        array_keys = ['seq_num', 'time', 'uid']
+        for i in range(num_pages):
+            yield {'descriptor': 'DESCRIPTOR',
+                   **{key: list(range(page_size)) for key in array_keys},
+                   'data': {key: list(range(page_size)) for key in data_keys},
+                   'timestamps': {key: list(range(page_size)) for key in data_keys},
+                   'filled': {key: list(range(page_size)) for key in data_keys}}
+
+    # Get a list of event pages of size 13.
+    event_pages = list(event_page_gen(13, 31))
+    # Change the size of the event_pages to size 7.
+    event_pages_7 = event_model.rechunk_event_pages(event_pages, 7)
```

Comment:
I think it would be good to check here that the sizes of the pages in `event_pages_7` are `57 * [4] + [4]` as expected. From interactive testing I believe this is working correctly, but for the sake of future-proofing against future changes we should verify that here.

Reply:
updated